File: core/logic.go
package core

import (
	"ddt-copilot/data"
	"ddt-copilot/defs"
	"ddt-copilot/utils"
	"github.com/lxn/win"
)

var (
	defaultAttackList = []string{"E", "2", "3",
		"4", "4", "4", "4", "4", "4", "4", "4",
		"5", "5", "5", "5", "5", "5", "5", "5",
		"6", "6", "6", "6", "6", "6", "6", "6",
		"7", "7", "7", "7", "7", "7", "7", "7",
		"8", "8", "8", "8", "8", "8", "8", "8"} // 默认的攻击指令
)

func GetAttackCMD() []uintptr {
	var cmds []uintptr
	attackCmd := data.GGameSetting.SettingGeneral.AttackCMD
	if len(attackCmd) == 0 {
		attackCmd = defaultAttackList
	}
	for _, s := range attackCmd {
		vk := defs.GetVkFromStr(s)
		cmds = append(cmds, vk)
	}
	return cmds
}

func UseSkillByConfig(hwnd win.HWND) {
	cmds := GetAttackCMD()
	for _, cmd := range cmds {
		utils.UseSkill(hwnd, cmd)
	}
}

func Launch(hwnd win.HWND, needAngle, power int) {
	num, err := data.GDefsAngle.GetAngle(hwnd)
	if err != nil {
		data.Log().Error().Timestamp().Timestamp().Err(err).Int("hwnd", int(hwnd)).Msg("launch failed")
	}
	diff := needAngle - num
	data.Log().Info().Timestamp().Int("hwnd", int(hwnd)).Int("captureNum", num).Int("diff", diff).
		Int("needAngle", needAngle).Int("power", power).Msg("launch")
	utils.UpdateAngle(hwnd, diff)
	utils.Launch(hwnd, power)
}

func OnBattleCustom(ctrl *ScriptCtrl) {
	utils.ConfirmDirection(ctrl.hwnd, data.GGameSetting.SettingFubenCustom.Direction)

	attackCmd := data.GGameSetting.SettingFubenCustom.AttackCMD
	if len(attackCmd) == 0 {
		attackCmd = defaultAttackList
	}
	for _, s := range attackCmd {
		vk := defs.GetVkFromStr(s)
		utils.UseSkill(ctrl.hwnd, vk)
	}

	Launch(ctrl.hwnd, data.GGameSetting.SettingFubenCustom.Angle, data.GGameSetting.SettingFubenCustom.Power)
}

func OnRoundInitMaYiGeneral(ctrl *ScriptCtrl) {

}

func OnBattleMaYiGeneral(ctrl *ScriptCtrl) {

}

File: backup/qt-c++/app/core/logic.cpp

//#include <string>
//#include <vector>
//#include <defs/constants.h>
//#include <utils/utils_ddt.h>
//#include <app/core/core.h>

//const std::vector<std::string> DefaultAttackList = {
//    "E", "2", "3",
//    "4", "4", "4", "4", "4", "4", "4", "4",
//    "5", "5", "5", "5", "5", "5", "5", "5",
//    "6", "6", "6", "6", "6", "6", "6", "6",
//    "7", "7", "7", "7", "7", "7", "7", "7",
//    "8", "8", "8", "8", "8", "8", "8", "8",
//};

///*
//    范围-based for 循环：简洁易用，适合大多数情况。
//    传统 for 循环：在需要索引或修改元素时使用。
//    迭代器：提供灵活的遍历方式，适合更复杂的场景。
//    std::for_each 和 Lambda：适合需要复杂操作的情况。
// */
//std::vector<uintptr_t> GetAttackCMD() {
//    std::vector<uintptr_t> vec;
//    // Todo：从全局配置中读取攻击指令
//    auto attackCMD = DefaultAttackList;
//    for (const auto& k : attackCMD) {
//        auto it = Constants::VKMap.find(k);
//        if (it == Constants::VKMap.end()) {
//            continue;
//        }
//        vec.push_back(it->second);
//    }
//    return vec;
//}

//void UseSkillByConfig(HWND hwnd) {
//    auto cmds = GetAttackCMD();
//    for (const auto& k : cmds) {
//        KeyBoard(hwnd, k);
//    }
//}

//void LaunchWithAP(HWND hwnd, int angle, int power) {
//    auto num = ScriptCtrlMgr::Instance()->GetAgnle(hwnd);
//    auto diff = angle - num;
//    UpdateAngle(hwnd, diff);
//    Launch(hwnd, power);
//}
